### PR TITLE
Merge gems

### LINKED ISSUE #133: [PROPOSAL] Combine gems, extract opensearch-aws-sigv4
### What are you proposing?

There are 5 gems in this repo:

- opensearch-api
- opensearch-dsl
- opensearch-ruby
- opensearch-aws-sigv4
- opensearch-transport

All except `opensearch-aws-sigv4` are always used by anyone wanting to call OpenSearch in Ruby.

- Propose to merge them into opensearch-ruby.
- Propose to extract opensearch-aws-sigv4 into a separate repo.

### What users have asked for this feature?

Coming from #127.  

### What problems are you trying to solve?

- I don't believe anyone ever uses opensearch-api, opensearch-dsl, opensearch-transport separately from opensearch-ruby. Seems like unnecessary overhead. 
- We have tags such as opensearch-api/2.1.2 and the number of tags will grow fast. Each tag only refers to a subset of the code, and may incorrectly imply some relationship between, say opensearch-ruby and opensearch-api in the tag. Theoretically they could even be incompatible.
- Our 1.x, 2.x branching strategy forces to release all the gems at the same time or create more confusion.

### What is the developer experience going to be?

Major increment opensearch-ruby and collapse the other 3 gems into it.

Extract code for opensearch-aws-sigv4 into a new repo called opensearch-ruby-aws-sigv4.

### PR DESCRIPTION
**Pull request by nhtruong**
### Description
- Removed sigv4 gem since it's already been moved to https://github.com/opensearch-project/opensearch-ruby-aws-sigv4
- Merged Lib files of the remaining 4 gems
- Merged Test and Spec files (With refactoring to the helper modules)
- Reworked all rake tasks:
  - Redid all Test tasks to fit single-gem-multi-module structure
  - Redid RDOC task to work with 1 gem instead of multiple
  - Removed unused tasks
- Updated all workflow to work with single-gem structure
- Added CHANGELOG.md and change_log workflow
- Fixed all rubocop infractions in `./lib/**/*`
- Updated all docs accordingly
- Cleaned up and reworked gemspec and gemfile
- Updated `build.sh` to work with single-gem structure.
- Reworked opensearch_ruby_console.

### Issues Resolved
Closes #133 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by nhtruong** [comment]
It's basically. Just waiting for https://github.com/opensearch-project/opensearch-ruby-aws-sigv4 to go public so the link checker workflow would pass. Will also have a summary of what was done.

**Comment by nhtruong** [comment]
This is ready for review. I've updated the description with all overview of what was done. Basically only one or two files were untouched while the rest were moved or modified 😅. So, take your time.

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -20,38 +20,60 @@ Fork [opensearch-project/opensearch-ruby](https://github.com/opensearch-project/
...
-export PORT=9200
-export DISABLE_SECURITY=true
-./../.github/actions/opensearch/run.sh
+
+Run unit tests of a specific module:
+```shell
+rake test:api:unit
+rake test:transport:unit
+rake test:dsl:unit
+rake test:client:unit
```
Comment (dblock) [inline DEVELOPER_GUIDE.md:56]: We can now add `rake test:unit`.

File: Rakefile [hunk 2]
```diff
@@ -24,108 +24,297 @@
...
-      puts
+    # TODO: Implement and Incorporate this task into github actions
+    desc 'Run API integration tests'
+    task integration: :integration_spec
+    RSpec::Core::RakeTask.new(:integration_spec) do |t|
+      t.pattern = 'spec/opensearch/api/**/*_integration_spec.rb' # Placeholder
+    end
+
+    desc 'Run API tests'
+    task :all do
```
Comment (dblock) [inline Rakefile:79]: Generally you can do `task name: [:prereq1, :prereq2] `.

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
Looks great, merge it!

**Review by VachaShah — APPROVED** [review]
LGTM!

**Pull request merged**